### PR TITLE
Create a filtermodel decorator

### PR DESCRIPTION
**Pull request by zachmullen**
@mgrauer PTALSTWYIO. If you are OK with how this looks, I will go ahead and convert all of our REST endpoints over. This will be a good exercise; in the folder resource alone I found two endpoints that were failing to filter their results.

### CODE REVIEW

**Comment by mgrauer** [comment]
> PTALSTWYIO

See if This is What You Intended Originally?

Such That Walrus Yawping Is Obstreperous?


Yes to both.  I like it.

**Comment by zachmullen** [comment]
> it seems like I am adding fields to a filter.

You are adding fields to the filter. The filter says what fields are allowed in the document, and you're adding other fields to that filter... maybe the word "filter" itself is a source of confusion here?

**Comment by mgrauer** [comment]
Indeed, I think of "filter" as what is removed. _.js uses "pluck" for the operation of passing through certain selected values.

I can live with it if the docstring clarified a bit more.

**Comment by zachmullen** [comment]
Ah I see. I'll add more to the docstring to indicate that this is the "whitelist" sort of filter. I'm used to thinking of a filter as specifying what passes through rather than what doesn't, but I can definitely see the ambiguity.

**Comment by zachmullen** [comment]
> PTALSTWYIO

Since this was your idea originally. I think I've addressed all your comments, so if you like what you see I will convert all the existing endpoints over.

**Comment by mgrauer** [comment]
LGTM.  Go for it.

:checkered_flag:

**Comment by zachmullen** [comment]
@girder/developers PTAL at this change, this will touch a lot of REST code. It's another step towards reducing a lot of our boilerplate into a set of decorator annotations.

**Comment by jcfr** [comment]
:+1:

**Comment by zachmullen** [comment]
Everything I care to convert for the moment has been converted.

**Comment by zachmullen** [comment]
Fixed.

File: girder/models/item.py [hunk 1]
```diff
@@ -339,6 +347,11 @@ def parentsToRoot(self, item, user=None, force=False):
 
         :param item: The item whose root to find
         :type item: dict
+        :param user: The user running the test (not required if force=True).
```
Comment (mgrauer) [inline girder/models/item.py]: `running the test` ?  seems out of place here

Comment (zachmullen) [inline girder/models/item.py]: This is one of those "did I really write that?" moments.

**Comment by mgrauer** [comment]
I'm wondering if there is an inconsistency in when you add the `access` to the filter, but looking through the code, I couldn't find any.  There's a lot of code though.  Can you summarize the policy for when to add `access` to the filter?

**Comment by zachmullen** [comment]
We more or less throw it into the document for any request where it could have changed. This minimizes the number of requests that client needs to make to the server. (At least, that's the goal... the execution might not be 100% correct.)

**Comment by mgrauer** [comment]
> Everything I care to convert for the moment has been converted.

    f(sizeof(codebase), sizeof(your_heart))

LGTM.
